src/processing/UserUtil.java

```diff
@@ -0,0 +1,141 @@
+package processing;
+
+import java.util.ArrayList;
+import processing.utility.*;
+import processing.server.board.*;
+import networking.utility.*;
+import networking.*;
+import infrastructure.validation.logger.*;
+
+/**
+ * Implements some of the User APIs
+ * 
+ * @author Rakesh Kumar
+ * @reviewer Devansh Singh Rathore
+ */
+
+public class UserUtil {
+	
+	// gets the logger instance
+	private static ILogger logger = LoggerFactory.getLoggerInstance();
+	
+	/**
+	 * UI calls this method to initiate connection 
+	 * with the server.
+	 * The information provided in the arguments
+	 * are used to start the connection
+	 * 
+	 * @param username Username
+	 * @param ipAddress Server's ip address
+	 * @param boardId Board ID	
+	 */
+    public static void initiateUserConnection(
+    		String username, 
+    		String ipAddress, 
+    		String boardId
+    ) {
+    	
+    	// stores the information
+    	ClientBoardState.username  = new Username(username);
```

Comment:
nested indentation varies by tab of width of minimum 4 spaces. Please check and correct it.

Reply:
....everywhere